lib/replaceWithDecompressedPrimitive.js

```diff
@@ -7,26 +14,165 @@ module.exports = replaceWithDecompressedPrimitive;
  *
  * @param {Object} gltf A javascript object containing a glTF asset.
  * @param {Object} primitive A javascript object containing a glTF primitive.
- * @param {Number} encodedPointsCount Number of points after the mesh is decompressed.
- * @param {Number} encodedFacesCount Number of faces after the mesh is decompressed.
+ * @param {Object} dracoEncodedBuffer A javascript object containing a Draco encoded mesh.
+ * @param {Number} [dracoEncodedBuffer.numberOfPoints] Number of points after the mesh is decompressed.
+ * @param {Number} [dracoEncodedBuffer.numberOfFaces] Number of faces after the mesh is decompressed.
+ * @param {Buffer} [dracoEncodedBuffer.buffer] A Buffer object containing a Draco compressed mesh.
+ * @param {Boolean} fallback If set, replaces the original with the decompressed data.
  * @returns {Object} The glTF asset with the decompressed primitive.
  *
  * @private
  */
-function replaceWithDecompressedPrimitive(gltf, primitive, encodedPointsCount, encodedFacesCount) {
+function replaceWithDecompressedPrimitive(gltf, primitive, dracoEncodedBuffer, fallback) {
+    var decoder;
+    var dracoGeometry;
+
     // Add decompressed indices data to indices accessor.
-    var indicesAccessorId = primitive.indices;
-    var indicesAccessor = gltf.accessors[indicesAccessorId];
-    indicesAccessor.count = encodedFacesCount * 3;
+    var indicesAccessor = gltf.accessors[primitive.indices];
+    indicesAccessor.count = dracoEncodedBuffer.numberOfFaces * 3;
+
+    if (fallback) {
+        decoder = new decoderModule.Decoder();
+        dracoGeometry = decompressDracoBuffer(decoder, dracoEncodedBuffer.buffer);
```

Comment:
What is the reason for replacing the existing uncompressed data with the decompressed data? Is it because the number of faces may have changed and the fallback data needs to match up?

Reply:
Yes, basically to make sure "count" in the accessors match.